lib/cocoapods/executable.rb

```diff
@@ -93,6 +94,21 @@ def self.popen3(bin, command, stdout, stderr)
       end
     end
 
+    def self.reader(input, output)
+      Thread.new do
+        buf = ''
+        begin
+          loop do
+            buf << input.readpartial(4096)
+            string, separator, buf = buf.rpartition(/[\r\n]/)
```

Comment:
Just try running the integration specs. I'm going to revert part of https://github.com/CocoaPods/CocoaPods/commit/4ee32880f40fbea417244568b8cdc76259be1975, because using `rpartition` here is incorrect -- we do need two loop.